debezium-connector-jdbc/src/main/java/io/debezium/connector/jdbc/type/connect/ConnectStringType.java

```diff
@@ -59,33 +60,25 @@ public String getTypeName(DatabaseDialect dialect, Schema schema, boolean key) {
             }
 
             if (columnSize > 0) {
-                return dialect.getTypeName(resolvedJdbcType, Size.length(columnSize));
+                return dialect.getJdbcTypeName(resolvedJdbcType, Size.length(columnSize));
             }
-            return dialect.getTypeName(resolvedJdbcType);
+            return dialect.getJdbcTypeName(resolvedJdbcType);
         }
         else {
             final int jdbcType = hasNationalizedCharacterSet(schema) ? Types.NVARCHAR : Types.VARCHAR;
-            // The column propagation details either don't exist or did not map to a logical STRING type
-            // In this case, we apply the size defaults (no size for non-keys and max-key for keys).
-            if (key) {
-                return dialect.getTypeName(jdbcType, Size.length(getMaxSizeInKey(dialect, jdbcType)));
-            }
-            return dialect.getTypeName(jdbcType);
+            return dialect.getJdbcTypeName(jdbcType, Size.length(getMaxSizeInKey(dialect, jdbcType)));
         }
     }
 
-    private int getColumnSize(DatabaseDialect dialect, Schema schema, boolean key, int jdbcType) {
+    private int getColumnSize(DatabaseDialect dialect, Schema schema, int jdbcType) {
         int columnSize = Integer.parseInt(getSourceColumnSize(schema).orElse("0"));
-        if (key) {
-            final int maxSizeInKey = getMaxSizeInKey(dialect, jdbcType);
-            if (columnSize > 0) {
-                columnSize = Math.min(columnSize, maxSizeInKey);
-            }
-            else {
-                columnSize = maxSizeInKey;
-            }
+        final int maxSizeInKey = getMaxSizeInKey(dialect, jdbcType);
```

Comment:
The max size in key should only be applied to key columns, not non-key columns.

Reply:
It's the same logic as before, only head and tail inverted, no?

Reply:
same

Reply:
With the new `isKey` argument, it is now :+1: